python-package/packager/nativelib.py

```diff
@@ -138,6 +138,9 @@ def locate_or_build_libxgboost(
             sys_prefix / "Library",
             sys_prefix / "Library" / "bin",
             sys_prefix / "Library" / "lib",
+            sys_prefix / "Library" / "mingw-w64",
+            sys_prefix / "Library" / "mingw-w64" / "bin",
```

Comment:
Could you please share under which condition libxgboost is on the bin path? Also, is mingw ever used for compiling the Python package? Asking since mingw has/had a broken Unix socket implementation, hence distributed computing is disabled there.

Reply:
Conda-forge uses Visual Studio to build the Python package. See the logs at https://github.com/conda-forge/xgboost-feedstock/runs/27951580159.

@jakirkham I am inclined to exclude this patch from the upstream because the use of `mingw-w64` appears to be a convention specific to Conda-forge and not generally applicable to all users.

Reply:
If conda forge uses visual studio, maybe we can simply drop the patch from conda forge as well?

Reply:
No, Conda-forge needs this patch. Libraries are installed under the mingw-w64 directory. I tried removing it and got errors.

Reply:
In the bin directory?

Reply:
Yes. The patch is a product of trial and error. I had to add it to get https://github.com/conda-forge/xgboost-feedstock/pull/132 working.

Reply:
That's unexpected. Thank you for sharing. I'm fine with merging it then.

Reply:
Thank you both! 🙏